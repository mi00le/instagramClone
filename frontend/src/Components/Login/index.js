import React, { Component } from "react";
import {Col} from "react-bootstrap";
import axios from "axios";
import qs from "qs";

const loginBox = {
  margin : "auto",
  float: "none",
  borderStyle: "solid",
  borderColor: "#e6e6e6",
  borderWidth: 1,
  borderRadius: 3,
  padding: "0",
  background: "#fff",
}

const loginInput = {
  width: "80%",
  height: 30,
  margin: "12px auto",
  padding: 2,
  display: "block",
  borderRadius: 3,
  borderWidth: 1,
  borderColor: "#e6e6e6",
}
const loginDiv = {
    position: "relative",
    height: "100vh",
    display: "flex",
    alignItems: "center",
}

export default class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      register: false,
      errorMsg: ""
    };

    this.swapSetting = this.swapSetting.bind(this);
    this.login = this.login.bind(this);
    this.register = this.register.bind(this);
  }

  swapSetting(){
    this.setState({
      register: !this.state.register,
      errorMsg: ""
    });
  }

  login(){
    console.log("login");
    axios.post("http://localhost:3002/users/auth", qs.stringify({email: this.state.email,password: this.state.password}))
    .then((res) => {
        if (!res.data.auth.success){throw new Error(res.data.auth.err.message)};
        this.props.sucessFunction(this.state.email, res.data.auth.user.displayName, res.data.auth.user.id, res.data.token);
        console.log("sucess");
    }).catch((err) => {
      console.log("err");
        this.setState({
          email: "",
          password: "",
          errorMsg: err.message,
          register: false
        });
    });
  }
  handleChange = (event) => {
    this.setState({email: event.target.value});
  }
  handleChangePass = (event) => {
    this.setState({password: event.target.value});
  }
  register(){
    const username = document.getElementById('username').value;
    const password2 = document.getElementById('password2').value;

    if (this.state.password != password2) {
      this.setState({
        errorMsg: "Passwords does not match",
      });
      return
    }
    axios.post("http://localhost:3002/users", qs.stringify({email: this.state.email,password: this.state.password, displayName: username}))
    .then((res) => {
      if(!res.data.data.success){throw new Error(res.data.data.err.message)};
      this.props.sucessFunction(this.state.email, username, res.data.data.user.id, res.data.token);
    }).catch((err) => {
        this.setState({
          errorMsg: err.message,
          register: true
        });
    });
  }

  render() {
    return (
      <div style={loginDiv}>
        <Col xs={12} sm={6} md={4} style={loginBox}>
          <h1>{this.state.register ? "Register" : "Login"}</h1>
          <hr />
          {this.state.errorMsg && <p className="ErrorMsg">{this.state.errorMsg}</p>}
          <form onSubmit={this.state.register ? this.register : this.login}>
            <h4>Email</h4>
            <input style={loginInput} value={this.state.value} onChange={this.handleChange} id="email" type="email" autoFocus/>
            {this.state.register && <h4>Username</h4>}
            {this.state.register && <input style={loginInput} id="username" type="text"/>}
            <h4>Password</h4>
            <input style={loginInput} value={this.state.value} onChange={this.handleChangePass} id="password" type="password"/>
            {this.state.register && <h4>Confirm password</h4>}
            {this.state.register && <input style={loginInput} id="password2" type="password"/>}
            <input style={loginInput} type="submit" value={this.state.register ? "Register" : "Login"} />
          <a style={{marginBottom: 10, display: "block"}} onClick={this.swapSetting}>{this.state.register ? "Already registered? Login!" : "New? Register!"}</a>
          </form>
        </Col>
      </div>
    );
  }
}
