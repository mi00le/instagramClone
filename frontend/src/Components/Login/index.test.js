import React from 'react'
import Login from './'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Login', () => {
  it('should render', () => {
    const wrapper = shallow(<Login />)
    expect(wrapper).toBeTruthy()
  })

  it('should handle login', () => {
    const login = jest.fn()
    Login.prototype.login = login;
    const wrapper = shallow(<Login />)
    wrapper.find('form').simulate('submit')
    expect(login).toHaveBeenCalled();
  })

  it('should match snapshot', () => {
    const tree = renderer
      .create(<Login />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
