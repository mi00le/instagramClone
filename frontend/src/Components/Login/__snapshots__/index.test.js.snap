// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "position": "relative",
    }
  }
>
  <div
    className="col-md-4 col-sm-6 col-xs-12"
    style={
      Object {
        "background": "#fff",
        "borderColor": "#e6e6e6",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1,
        "float": "none",
        "margin": "auto",
        "padding": "0",
      }
    }
  >
    <h1>
      Login
    </h1>
    <hr />
    
    <form
      onSubmit={[Function]}
    >
      <h4>
        Email
      </h4>
      <input
        autoFocus={true}
        id="email"
        onChange={[Function]}
        style={
          Object {
            "borderColor": "#e6e6e6",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "block",
            "height": 30,
            "margin": "12px auto",
            "padding": 2,
            "width": "80%",
          }
        }
        type="email"
      />
      <h4>
        Password
      </h4>
      <input
        id="password"
        onChange={[Function]}
        style={
          Object {
            "borderColor": "#e6e6e6",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "block",
            "height": 30,
            "margin": "12px auto",
            "padding": 2,
            "width": "80%",
          }
        }
        type="password"
      />
      <input
        style={
          Object {
            "borderColor": "#e6e6e6",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "block",
            "height": 30,
            "margin": "12px auto",
            "padding": 2,
            "width": "80%",
          }
        }
        type="submit"
        value="Login"
      />
      <a
        onClick={[Function]}
        style={
          Object {
            "display": "block",
            "marginBottom": 10,
          }
        }
      >
        New? Register!
      </a>
    </form>
  </div>
</div>
`;
